### PR TITLE
DM-23985: Cannot do linearity corrections in Gen 3 DECam processing

### PR DESCRIPTION
**Pull request by czwa**
Add linearizer connection for gen3.

### CODE REVIEW

**Review by mrawls — COMMENTED** [review]
Needs some tests for the various flavors of linearizer inputs, otherwise looks OK.

File: python/lsst/ip/isr/linearize.py [hunk 1]
```diff
@@ -339,7 +339,7 @@ def toTable(self):
         catalog.meta = self.getMetadata().toDict()
         tableList.append(catalog)
 
-        if self.tableData:
+        if self.tableData is not None:
```
[the remark this answers is not in the document]
Comment (czwa) [reply python/lsst/ip/isr/linearize.py:342]: I forgot that populated numpy arrays throw ValueError when used alone (the "The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()" error).

File: python/lsst/ip/isr/isrTask.py [hunk 2]
```diff
@@ -942,13 +949,20 @@ def runQuantum(self, butlerQC, inputRefs, outputRefs):
                     self.log.warn("No crosstalkSources found for chip with interChip terms!")
 
         if self.doLinearize(detector) is True:
-            if 'linearizer' in inputs and isinstance(inputs['linearizer'], dict):
-                linearizer = linearize.Linearizer(detector=detector, log=self.log)
-                linearizer.fromYaml(inputs['linearizer'])
+            if 'linearizer' in inputs:
```
[the remark this answers is not in the document]
Comment (czwa) [reply python/lsst/ip/isr/isrTask.py:952]: This is an unfortunate case of trying not to break existing workflows.  I've filed a ticket (DM-28741) that will remove support for the "old-style" linearizers, leaving only the ip_isr calibration type and the cameraGeom as possible sources of linearity information.  I've added log warnings in this block to note this (and to provide myself ways to check that no camera or butler triggers when I work on that ticket).

[the remark this answers is not in the document]
Comment (mrawls) [reply python/lsst/ip/isr/isrTask.py:952]: OK, thanks for the warnings, that's helpful. I don't have a good sense for which existing workflows use these older style of linearizer. You've manually checked to see that both work like this? Can you add a test just for the new/preferred option?

File: python/lsst/ip/isr/linearize.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mrawls) [reply python/lsst/ip/isr/linearize.py:342]: Got it. So we just got lucky this didn't break before?

File: tests/test_linearize.py [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+                storedLinearizer = Linearizer().fromTable(storedTable)
+
+                measImage = inImage.Factory(inImage, True)
+                storedLinearizer = self.makeLinearizer(linearityType)
+                storedResult = storedLinearizer.applyLinearity(measImage, log=self.log)
+
+                self.assertImagesAlmostEqual(refImage, measImage)
+                self.assertEqual(storedResult.numOutOfRange, refNumOutOfRange)
+                self.assertEqual(storedResult.numLinearized, self.numAmps - zeroLinearity)
+                self.assertEqual(storedResult.numAmps, self.numAmps)
```
Comment (mrawls) [inline tests/test_linearize.py:206]: Can this bit be its own function since you do it 3x?

File: tests/test_linearize.py [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+
+    def makeDetector(self, linearityType, bbox=None):
+        """
+        """
```
Comment (mrawls) [inline tests/test_linearize.py:210]: Please fill in the blank 🙂 both here and in `makeLinearizer`

**Review by mrawls — APPROVED** [review]
Thanks for adding the tests!!

**Pull request merged**